srcCpp/perftest_publisher.cxx

```diff
@@ -1783,9 +1780,50 @@ int perftest_cpp::Publisher()
 
     message.size = INITIALIZE_SIZE;
 
-    for (unsigned long i = 0;
-            i < (std::max)(_InstanceCount, announcementSampleCount);
-            i++) {
+    /*
+     * The purpose of this initial burst of Data is to make sure that all the
+     * queues are initialized already by the time the test starts taking
```

Comment:
Ensure that most Memory allocations are done in the critical path before the test starts.

Also that Instances are registered in advance.